README.md
# Anastasia.com

history
    1  ls
    2  cd ./
    3  cd /
    4  ls
    5  curl -fsSL https://deb.nodesource.com/setup_17.x | sudo -E bash -
    6  apt install curl
    7  clear
    8  ls
    9  curl -fsSL https://deb.nodesource.com/setup_17.x | sudo -E bash -
   10  node -v
   11  curl -fsSL https://deb.nodesource.com/setup_17.x | sudo -E bash -
   12  sudo apt-get install -y nodejs
   13  node -v
   14  npm -v
   15  apt install git
   16  ls
   17  cd home
   18  mkdir apps
   19  cd apps
   20  git clone https://github.com/Fapsap/Anastasia.com.git
   21  ls
   22  cd Anastasia.com/
   23  ls
   24  npm i
   25  apt install nginx
   26  node index.js
   27  sudo nano /etc/nginx/sites-available/default
   28  nginx -t
   29  sudo nano /etc/nginx/sites-available/default
   30  nginx -t
   31  systemctl restart nginx
   32  node index.js
   33  sudo npm install pm2 -g
   34  pm2 start index.js
   35  pm2 --help
   36  pm2 startup index.js
   37  pm2 startup
   38  cd /etc/systemd/system
   39  ls
   40  sudo nano /etc/nginx/sites-available/default
   41  apt show
   42  apt list
   43  dpkg -l
   44  dpkg -l > pkg.txt
   45  cat pkg.txt
   46  history > history


index.js
import express from "express";
import path from 'path';
import bodyParser from "body-parser";
import TelegramBot from "node-telegram-bot-api";

const token = '2131814982:AAEjqyxZz0hy0I19bm6daMvdhIrqsFcJNXk'
const bot = new TelegramBot(token, { polling: true })

const __dirname = path.resolve()

const PORT = process.env.PORT ?? 3000
const app = express()

app.use(express.static(path.resolve(__dirname, 'public')))
app.use(bodyParser.urlencoded({ extended: true }));
	
const jsonParser = express.json();
  
app.post("/index.html", jsonParser, function (request, response) {
    console.log(request.body);
    if(!request.body) return response.sendStatus(400);
    let tg_msg = 'Почта: ' + request.body.userEmail + "\nФИО: " + request.body.userFio + "\nТема сообщения: " + request.body.TemaMsg  + "\nТекст сообщения: \n" + request.body.MsgText;

    bot.sendMessage('816283979', tg_msg)

    response.json(request.body); 
    
});



app.listen(PORT)

public/scripts/Postform.js
btn = document.getElementsByClassName("form_btn")
form = document.getElementsByClassName('form')

inputemail = document.getElementById("email")

document.forms.connection_form.onsubmit = function(e){
    e.preventDefault();

    //var userEmail = document.forms.connection_form.email.value;
    //var userFio = document.forms.connection_form.fio.value;
    //var TemaMsg = document.forms.connection_form.t_txt.value;
    //var MsgText = document.forms.connection_form.msg_txt.value;

    //var FormValue = "Почта: "  + userEmail + "\nФИО: " + userFio + "\nТема сообщения:" + TemaMsg + "\nСообщение:\n" + MsgText

    //var xhr = new XMLHttpRequest();

    //xhr.open('POST', '/index.html');
    //xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded')

    //xhr.send(FormValue)

    let registerForm = document.forms["connection_form"];
    
    let userEmail = connection_form.elements["email"].value;
    let userFio = registerForm.elements["fio"].value;
    let TemaMsg = registerForm.elements["t_txt"].value;
    let MsgText = registerForm.elements["msg_txt"].value;
    
    console.log("собрали инфу")

            // сериализуем данные в json
    let user = JSON.stringify({userEmail: userEmail, userFio: userFio, TemaMsg: TemaMsg, MsgText: MsgText});
    let request = new XMLHttpRequest();

            // посылаем запрос на адрес "/user"
    request.open("POST", "/index.html", true);   
    request.setRequestHeader("Content-Type", "application/json");
    request.addEventListener("load", function () {
                // получаем и парсим ответ сервера
        let receivedUser = JSON.parse(request.response);
        console.log(receivedUser.userEmail, receivedUser.userFio);   // смотрим ответ сервера
    });
    
    request.send(user);
    console.log("Отправили инфу")
};

public/scripts/burger-menu.js
let menuBtn = document.querySelector('.menu-btn');
let menu = document.querySelector('.menu');

menuBtn.addEventListener('click', function(){
	menuBtn.classList.toggle('active');
	menu.classList.toggle('active');
})

let conn_btn = document.querySelector('.m-conn-btn');
conn_btn.addEventListener('click', function(){
	menuBtn.classList.toggle('active');
	menu.classList.toggle('active');
})

let currentUrl = window.location.pathname;
console.log(currentUrl)

let d_m = document.getElementById('d_href_main');
let d_a = document.getElementById('d_href_about');
let d_f = document.getElementById('d_href_faq');
let d_r = document.getElementById('d_href_rev');
let d_art = document.getElementById('d_href_art');


if (currentUrl == "/index.html" || currentUrl == "/"){
	d_m.style.borderBottom = "1px solid #000"
}
else if (currentUrl == "/about.html"){
	d_a.style.borderBottom = "1px solid #000"
}
else if (currentUrl == "/faq.html"){
	d_f.style.borderBottom = "1px solid #000"
}
else if (currentUrl == "/reviews.html"){
	d_r.style.borderBottom = "1px solid #000"
}
else if (currentUrl == "/articles.html"){
	d_art.style.borderBottom = "1px solid #000"
}

let m_m = document.getElementById('m_href_main');
let m_a = document.getElementById('m_href_about');
let m_f = document.getElementById('m_href_faq');
let m_r = document.getElementById('m_href_rev');
let m_art = document.getElementById('m_href_art');

if (currentUrl == "/index.html" || currentUrl == "/"){
	m_m.style.backgroundColor = "#DB4BFF";
	m_m.style.color = "#fff"
}
else if (currentUrl == "/about.html"){
	m_a.style.backgroundColor = "#DB4BFF";
	m_a.style.color = "#fff"
}
else if (currentUrl == "/faq.html"){
	m_f.style.backgroundColor = "#DB4BFF";
	m_f.style.color = "#fff"
}
else if (currentUrl == "/reviews.html"){
	m_r.style.backgroundColor = "#DB4BFF";
	m_r.style.color = "#fff"
}
else if (currentUrl == "/articles.html"){
	m_art.style.backgroundColor = "#DB4BFF";
	m_art.style.color = "#fff";
}

//d_m.getAttribute("href")
let form_сonn = document.getElementsByClassName("form");

function btn_send_q() {
    //let form_сonn = document.querySelector('.form')[0];
	//form_conn.setAttribute("disabled", "");
	setCookie('sndform', 'true', {expires: Date('2030-12-17T03:24:00')});
	localStorage.setItem('sndform', 'true');
	document.getElementById('form_btn').setAttribute("disabled", "");
}

function btn_send(){
	if (getCookie('sndform') == 'true' || localStorage.sndform == 'true'){
		document.getElementById('form_btn').setAttribute("value", "ОТПРАВЛЕНО!");
	}
	else{
		document.getElementById('form_btn').setAttribute("value", "ОТПРАВЛЕНО!");
		setTimeout(btn_send_q, 1500);
	}
}
var h_hght = 25; // высота шапки
        var h_mrg = 10;     // отступ когда шапка уже не видна
        $(function(){
         $(window).scroll(function(){
            var top = $(this).scrollTop();
            var elem = $('#menu-btn');
            if (top+h_mrg < h_hght) {
             elem.css('top', (h_hght-top));
            } else {
             elem.css('top', h_mrg);
            }
          });
        });
